packages/jest-reporters/src/__tests__/get_result_header.test.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {makeGlobalConfig} from '../../../../TestUtils';
+import {formatTestPath} from '../utils';
+import getResultHeader from '../get_result_header';
+const terminalLink = require('terminal-link');
+
+jest.mock('terminal-link', () => jest.fn(() => 'wannabehyperlink'));
+
+const testResult = {
+  testFilePath: '/foo',
+};
+
+const globalConfig = makeGlobalConfig();
+
+test('should call `terminal-link` correctly', () => {
+  terminalLink.mockClear();
+
+  getResultHeader(testResult, globalConfig);
+  const call = terminalLink.mock.calls[0];
+
+  expect(terminalLink).toHaveBeenCalled();
+  expect(call[0]).toBe(formatTestPath(globalConfig, testResult.testFilePath));
```

Comment:
Would prefer if we hardcode the expected call argument values here without using `formatTestPath`/`testResult`, it kinda reimplements the production code this way and is not easy on the eye.
Also I think `toHaveBeenCalledWith` simplifies this a lot (4 lines to 1)?

Reply:
I've tried hardcoding the `formatTestPath` result by copy-pasting it from the console, but it contains the parts which are coloured by the chalk library and it seems to get lost when copied, any idea how could I keep it if we still want to go hardcode it?
btw. is reimplementing the production code in tests a bad practice? if so, what's exactly wrong with it?

Reply:
Could mock chalk like some other tests do so it just becomes something like `<green>text</green>`. But that's probably too complicated. Maybe just `expect.stringContaining('the test path')`?
What I mean re reimplementing: The reason we write tests is to run the code with specific example data that reproduces a case simple enough to wrap our heads around. If we could manage the complexity of all cases in all the code at once, we wouldn't need tests but could just look at the production code and see whether it's correct ;)

Reply:
That makes sense, thanks for the explanation @jeysal ! I've hardcoded the expected values, let me know if that's what you had in mind :)